File: Back/CompanyService/src/main/java/com/StockMarketCharting/CompanyService/model/Company.java
package com.StockMarketCharting.CompanyService.model;

import java.util.ArrayList;
import java.util.List;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinTable;
import javax.persistence.ManyToMany;
import javax.persistence.ManyToOne;
import javax.persistence.OrderColumn;
import javax.persistence.Table;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Company {

	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
	private Integer id;
	
	@Column(nullable = false)
	private String comapnyName;
	
	@ManyToOne
	private Sector sector;
	
	@ManyToMany(cascade = CascadeType.ALL)
	@JoinTable(name = "company_stockExchanges")
	private List<StockExchange> stockExhanges = new ArrayList<>();
	
	private String ceo;
	
	private Integer turnover;
	
	private String[] boardOfDirectors;
	
	private String details;
	
	@Column(nullable = false)
	private String stockCode;
	
}
/*
5. Listed in Stock Exchanges
6. Sector
8. Stock code in each Stock Exchange
*/

File: Front/angularapp/src/app/models/Company.ts
import { Sector } from "./Sector";
import { StockExchange } from "./StockExchange";

export interface Company{
    id:string;
    companyName: string;
    turnover?: number;
    ceo?: string;
    boardOFDirectors?: string[];
    stockExchanges?: StockExchange[];
    sector?: Sector;
    stockCode?: string;
    brief?: string;
}
/*
2. Turnover
3. CEO
4. Board of Directors
5. Listed in Stock Exchanges
6. Sector
7. Brief writeup, about companies Services/Product, etc…
8. Stock code in each Stock Exchange
 */